### PR TITLE
Node-killer avoid killing nodes where prometheus is running

### PR DESCRIPTION
**Pull request by mozhuli**
Ref #777 

/assign @mm4tt

### CODE REVIEW

File: clusterloader2/pkg/framework/client/objects.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ const (
 	// Parameters for namespace deletion operations.
 	defaultNamespaceDeletionTimeout  = 10 * time.Minute
 	defaultNamespaceDeletionInterval = 5 * time.Second
+
+	namespace       = "monitoring"
```
Comment (wojtek-t) [inline clusterloader2/pkg/framework/client/objects.go]: Please name it in more descriptive way, e.g. monitoringNamespace

Comment (wojtek-t) [reply clusterloader2/pkg/framework/client/objects.go]: Actually, let's remove it completely from this file - it shouldn't live here (see also my comment below).

File: clusterloader2/pkg/framework/client/objects.go [hunk 2]
```diff
@@ -131,6 +134,30 @@ func RetryFunction(f func() error, options ...*ApiCallOptions) wait.ConditionFun
 	}
 }
 
+// ListPrometheusPods returns list of prometheus pods.
+func ListPrometheusPods(c clientset.Interface) ([]apiv1.Pod, error) {
```
Comment (wojtek-t) [inline clusterloader2/pkg/framework/client/objects.go]: Let's remove this function and call ListPodsWithOptions where you need it.

File: clusterloader2/pkg/framework/client/objects.go (hunk 1 above)
Comment (mozhuli) [reply clusterloader2/pkg/framework/client/objects.go]: @wojtek-t updated!

File: clusterloader2/pkg/util/cluster.go [hunk 3]
```diff
@@ -38,9 +45,22 @@ func GetSchedulableUntainedNodes(c clientset.Interface) ([]corev1.Node, error) {
 	if err != nil {
 		return nil, err
 	}
+	podList, err := client.ListPodsWithOptions(c, monitoringNamespace, metav1.ListOptions{
```
Comment (wojtek-t) [inline clusterloader2/pkg/util/cluster.go]: nit:
s/podList/prometheusPods/

Comment (mozhuli) [reply clusterloader2/pkg/util/cluster.go]: done

File: clusterloader2/pkg/util/cluster.go [hunk 4]
```diff
@@ -38,9 +45,22 @@ func GetSchedulableUntainedNodes(c clientset.Interface) ([]corev1.Node, error) {
 	if err != nil {
 		return nil, err
 	}
+	prometheusPods, err := client.ListPodsWithOptions(c, monitoringNamespace, metav1.ListOptions{
```
Comment (wojtek-t) [inline clusterloader2/pkg/util/cluster.go]: I'm sorry, I didn't realize that before.
This logic shouldn't go to GetUnschedulableUntaintedNodes - this is very generic function that is used in many context.

This logic should go directly to chaos monkey code here:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/chaos/nodes.go#L67

Comment (mm4tt) [reply clusterloader2/pkg/util/cluster.go]: +1

File: clusterloader2/pkg/util/cluster.go [hunk 5]
```diff
@@ -19,11 +19,12 @@ package util
 import (
 	"fmt"
 
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
```
Comment (wojtek-t) [inline clusterloader2/pkg/util/cluster.go]: last nit - please revert this line - we try to keep imports in alphabetical order, so the previous one was good

File: clusterloader2/pkg/util/cluster.go (hunk 4 above)
Comment (mozhuli) [reply clusterloader2/pkg/util/cluster.go]: @mm4tt @mm4tt Have fixed!  Sorry for not noticing this

File: clusterloader2/pkg/util/cluster.go (hunk 5 above)
Comment (mozhuli) [reply clusterloader2/pkg/util/cluster.go]: not the `go package->this project package->vendor package`?

**Comment by wojtek-t** [comment]
/lgtm
/approve

@mozhuli - thanks a lot for pushing this - it's super useful!

**Comment by mm4tt** [comment]
/lgtm

Thanks!

**Comment by mm4tt** [comment]
Looks like the gofmt failed for one file:

```
$ ./verify/verify-gofmt.sh
!!! 'gofmt -s' needs to be run on the following files: 
./clusterloader2/pkg/chaos/nodes.go
The command "./verify/verify-gofmt.sh" exited with 1.
```

Please fix

**Comment by mozhuli** [comment]
> Looks like the gofmt failed for one file:
> 
> ```
> $ ./verify/verify-gofmt.sh
> !!! 'gofmt -s' needs to be run on the following files: 
> ./clusterloader2/pkg/chaos/nodes.go
> The command "./verify/verify-gofmt.sh" exited with 1.
> ```
> 
> Please fix

Fixed! Sorry for forgetting to format it.

**Comment by wojtek-t** [comment]
/lgtm
/approve

@mozhuli - thanks a lot!

**Pull request merged**